utils/div_ceil.hh

```diff
@@ -9,8 +9,18 @@
 #pragma once
 
 #include <concepts>
+#include <limits>
+#include "utils/assert.hh"
 
+template <std::integral Divdend, std::integral Divisor>
```

Comment:
Question: Is there a reason why we don't use `std::unsigned_integral` instead of `std::integral` here? I've just checked and we don't use the function in that many places, most of which seem to already use unsigned integer types after skimming through the code.